schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AbstractAvroSnapshotDeserializer.java

```diff
@@ -159,4 +166,39 @@ protected Object buildDeserializedObject(InputStream payloadInputStream,
 
         return deserializedObj;
     }
+
+
+    private DatumReader getDatumReader(Schema writerSchema, Schema readerSchema) {
+        if(this.useSpecificAvroReader) {
+            if(readerSchema == null) {
+                readerSchema = this.getReaderSchema(writerSchema);
+            }
+
+            return new SpecificDatumReader(writerSchema, readerSchema);
+        } else {
+            return readerSchema == null?new GenericDatumReader(writerSchema):new GenericDatumReader(writerSchema, readerSchema);
+        }
+    }
+
+    private Schema getReaderSchema(Schema writerSchema) {
+        Schema readerSchema = this.readerSchemaCache.get(writerSchema.getFullName());
+        if(readerSchema == null) {
+            Class readerClass = SpecificData.get().getClass(writerSchema);
+            if(readerClass == null) {
+                throw new SerializationException("Could not find class " + writerSchema.getFullName() + " specified in writer\'s schema whilst finding reader\'s schema for a SpecificRecord.");
```

Comment:
We may want to use `SerDesException` here. Better not to use kafka specific `SerializationException` as this class is for avro, not specifically for kafka.